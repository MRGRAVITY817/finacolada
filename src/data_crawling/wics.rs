use serde_json::Value;

use {super::biz_day::get_latest_biz_day, serde::Deserialize};

#[derive(Deserialize, Debug)]
#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
pub struct WicsRow {
    idx_cd: String,
    idx_nm_kor: String,
    all_mkt_val: u64,
    cmp_cd: String,
    cmp_kor: String,
    mkt_val: u64,
    wgt: f32,
    s_wgt: f32,
    cal_wgt: f32,
    sec_cd: String,
    sec_nm_kor: String,
    seq: i32,
    top60: u32,
    apt_shr_cnt: u64,
}

pub const SECTOR_CODE_LIST: [&str; 10] = [
    "G25", "G35", "G50", "G40", "G10", "G20", "G55", "G30", "G15", "G45",
];

pub async fn get_wics(
    query_client: &reqwest::Client,
    date: &str,
    sector_code: &str,
) -> Result<String, reqwest::Error> {
    // 1. Request the WICS data with URL & params
    let url = "https://www.wiseindex.com/Index/GetIndexComponets";
    query_client
        .get(url)
        .query(&[("ceil_yn", "0"), ("dt", date), ("sec_cd", sector_code)])
        .send()
        .await?
        .text()
        .await
}

pub async fn get_daily_wics_list(
    query_client: &reqwest::Client,
    date: &str,
) -> anyhow::Result<Vec<WicsRow>> {
    let mut row_list: Vec<WicsRow> = vec![];
    for sector_code in SECTOR_CODE_LIST {
        let json_string = get_wics(query_client, date, sector_code).await?;
        let meta: Value = serde_json::from_str(&json_string)?;
        for wics in meta["list"].as_array().unwrap() {
            let item: WicsRow = serde_json::from_value(wics.to_owned())?;
            row_list.push(item);
        }
    }

    Ok(row_list)
}

// pub async fn get_latest_wisc_data(query_client: &reqwest::Client) -> anyhow::Result<String> {
//     // 1. Get WICS data for every sector codes
//     let date = get_latest_biz_day(query_client).await?;
//     // 2. Deserialize JSON string as struct
//     let meta: Value = serde_json::from_str(&)
//     // 3. Get the "list" property
//     // 4. Create a DataFrame that contains each sector items.
//     let df = df![
//       "IDX_CD" => idx_cd,
//       "IDX_NM_KOR" => idx_nm_kor,
//       "ALL_MKT_VAL" => all_mkt_val,
//       "CMP_CD" => cmp_cd,
//       "CMP_KOR" => cmp_kor,
//       "MKT_VAL" => mkt_val,
//       "WGT" => wgt,
//       "S_WGT" => s_wgt,
//       "CAL_WGT" => cal_wgt,
//       "SEC_CD" => sec_cd,
//       "SEC_NM_KOR" => sec_nm_kor,
//       "SEQ" => seq,
//       "TOP60" => top60,
//       "APT_SHR_CNT" => apt_shr_cnt
//     ]?;
//     // 5. Save DataFrame as parquet file
//     let file_name = "assets/wics/latest_wics_data.parquet";
//     // 6. Return the parquet file name
//     Ok(file_name.to_string())
// }

#[cfg(test)]
mod test {
    use {super::*, insta::assert_yaml_snapshot};

    #[tokio::test]
    async fn should_have_json_text() {
        // Arrange
        let client = reqwest::Client::new();
        let date = "20230126";
        let sector_code = "G10";
        // Act
        let result = get_wics(&client, date, sector_code).await.unwrap();
        // Assert
        assert_yaml_snapshot!(result)
    }

    #[tokio::test]
    async fn different_date_should_have_different_data() {
        // Arrange
        let client = reqwest::Client::new();
        let early_date = "20230126";
        let now_date = "20230127";
        let sector_code = "G10";
        // Act
        let early_result = get_wics(&client, early_date, sector_code).await.unwrap();
        let now_result = get_wics(&client, now_date, sector_code).await.unwrap();
        // Assert
        assert_ne!(early_result, now_result)
    }

    #[tokio::test]
    async fn gets_wics_rows() {
        // Arrange
        let client = reqwest::Client::new();
        let date = "20230126";
        // Act
        let result = get_daily_wics_list(&client, date).await.unwrap();
        // Assert
        assert_yaml_snapshot!(format!("{:?}", result.iter().next()))
    }
}
