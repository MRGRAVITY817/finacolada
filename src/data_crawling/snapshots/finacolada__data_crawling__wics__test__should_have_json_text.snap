---
source: src/data_crawling/wics.rs
expression: result
---
"{\"info\":{\"TRD_DT\":\"\\/Date(1674658800000)\\/\",\"MKT_VAL\":21370791,\"TRD_AMT\":259421,\"CNT\":35},\"list\":[{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"096770\",\"CMP_KOR\":\"SK이노베이션\",\"MKT_VAL\":8512750,\"WGT\":39.83,\"S_WGT\":39.83,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":1,\"TOP60\":3,\"APT_SHR_CNT\":51780716},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"010950\",\"CMP_KOR\":\"S-Oil\",\"MKT_VAL\":3678192,\"WGT\":17.21,\"S_WGT\":57.04,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":2,\"TOP60\":3,\"APT_SHR_CNT\":41655633},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"267250\",\"CMP_KOR\":\"HD현대\",\"MKT_VAL\":2795723,\"WGT\":13.08,\"S_WGT\":70.13,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":3,\"TOP60\":3,\"APT_SHR_CNT\":44236128},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"078930\",\"CMP_KOR\":\"GS\",\"MKT_VAL\":2225881,\"WGT\":10.42,\"S_WGT\":80.54,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":4,\"TOP60\":3,\"APT_SHR_CNT\":49245150},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"112610\",\"CMP_KOR\":\"씨에스윈드\",\"MKT_VAL\":1612972,\"WGT\":7.55,\"S_WGT\":88.09,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":5,\"TOP60\":3,\"APT_SHR_CNT\":23615986},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"006120\",\"CMP_KOR\":\"SK디스커버리\",\"MKT_VAL\":329307,\"WGT\":1.54,\"S_WGT\":89.63,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":6,\"TOP60\":3,\"APT_SHR_CNT\":10470820},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"322000\",\"CMP_KOR\":\"현대에너지솔루션\",\"MKT_VAL\":264600,\"WGT\":1.24,\"S_WGT\":90.87,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":7,\"TOP60\":3,\"APT_SHR_CNT\":5040000},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"005090\",\"CMP_KOR\":\"SGC에너지\",\"MKT_VAL\":240541,\"WGT\":1.13,\"S_WGT\":91.99,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":8,\"TOP60\":3,\"APT_SHR_CNT\":7925582},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"018000\",\"CMP_KOR\":\"유니슨\",\"MKT_VAL\":186442,\"WGT\":0.87,\"S_WGT\":92.87,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":9,\"TOP60\":3,\"APT_SHR_CNT\":106355780},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"002960\",\"CMP_KOR\":\"한국쉘석유\",\"MKT_VAL\":138437,\"WGT\":0.65,\"S_WGT\":93.51,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":10,\"TOP60\":3,\"APT_SHR_CNT\":598000},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"126720\",\"CMP_KOR\":\"수산인더스트리\",\"MKT_VAL\":129767,\"WGT\":0.61,\"S_WGT\":94.12,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":11,\"TOP60\":3,\"APT_SHR_CNT\":5285820},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"126880\",\"CMP_KOR\":\"제이엔케이히터\",\"MKT_VAL\":113634,\"WGT\":0.53,\"S_WGT\":94.65,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":12,\"TOP60\":3,\"APT_SHR_CNT\":19260069},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"099220\",\"CMP_KOR\":\"SDN\",\"MKT_VAL\":96736,\"WGT\":0.45,\"S_WGT\":95.11,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":13,\"TOP60\":3,\"APT_SHR_CNT\":49992912},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"046120\",\"CMP_KOR\":\"오르비텍\",\"MKT_VAL\":96690,\"WGT\":0.45,\"S_WGT\":95.56,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":14,\"TOP60\":3,\"APT_SHR_CNT\":20572306},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"044490\",\"CMP_KOR\":\"태웅\",\"MKT_VAL\":91850,\"WGT\":0.43,\"S_WGT\":95.99,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":15,\"TOP60\":3,\"APT_SHR_CNT\":9203395},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"137950\",\"CMP_KOR\":\"제이씨케미칼\",\"MKT_VAL\":89160,\"WGT\":0.42,\"S_WGT\":96.41,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":16,\"TOP60\":3,\"APT_SHR_CNT\":12247298},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"060900\",\"CMP_KOR\":\"대한그린파워\",\"MKT_VAL\":88465,\"WGT\":0.41,\"S_WGT\":96.82,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":17,\"TOP60\":3,\"APT_SHR_CNT\":146952112},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"270520\",\"CMP_KOR\":\"지엔원에너지\",\"MKT_VAL\":88044,\"WGT\":0.41,\"S_WGT\":97.23,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":18,\"TOP60\":3,\"APT_SHR_CNT\":20595028},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"012320\",\"CMP_KOR\":\"경동인베스트\",\"MKT_VAL\":87908,\"WGT\":0.41,\"S_WGT\":97.64,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":19,\"TOP60\":3,\"APT_SHR_CNT\":993310},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"091090\",\"CMP_KOR\":\"세원이앤씨\",\"MKT_VAL\":81762,\"WGT\":0.38,\"S_WGT\":98.03,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":20,\"TOP60\":3,\"APT_SHR_CNT\":151130894},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"083650\",\"CMP_KOR\":\"비에이치아이\",\"MKT_VAL\":78021,\"WGT\":0.37,\"S_WGT\":98.39,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":21,\"TOP60\":3,\"APT_SHR_CNT\":11440000},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"128820\",\"CMP_KOR\":\"대성산업\",\"MKT_VAL\":70217,\"WGT\":0.33,\"S_WGT\":98.72,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":22,\"TOP60\":3,\"APT_SHR_CNT\":15832417},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"014530\",\"CMP_KOR\":\"극동유화\",\"MKT_VAL\":65799,\"WGT\":0.31,\"S_WGT\":99.03,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":23,\"TOP60\":3,\"APT_SHR_CNT\":17783404},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"389260\",\"CMP_KOR\":\"대명에너지\",\"MKT_VAL\":61892,\"WGT\":0.29,\"S_WGT\":99.32,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":24,\"TOP60\":3,\"APT_SHR_CNT\":2557500},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"093230\",\"CMP_KOR\":\"이아이디\",\"MKT_VAL\":58764,\"WGT\":0.27,\"S_WGT\":99.59,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":25,\"TOP60\":3,\"APT_SHR_CNT\":65439307},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"024060\",\"CMP_KOR\":\"흥구석유\",\"MKT_VAL\":55175,\"WGT\":0.26,\"S_WGT\":99.85,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":26,\"TOP60\":3,\"APT_SHR_CNT\":9150000},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"095910\",\"CMP_KOR\":\"에스에너지\",\"MKT_VAL\":50687,\"WGT\":0.24,\"S_WGT\":100.09,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":27,\"TOP60\":3,\"APT_SHR_CNT\":15018467},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"013810\",\"CMP_KOR\":\"스페코\",\"MKT_VAL\":46864,\"WGT\":0.22,\"S_WGT\":100.31,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":28,\"TOP60\":3,\"APT_SHR_CNT\":9672610},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"032980\",\"CMP_KOR\":\"바이온\",\"MKT_VAL\":41471,\"WGT\":0.19,\"S_WGT\":100.50,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":29,\"TOP60\":3,\"APT_SHR_CNT\":32348925},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"038870\",\"CMP_KOR\":\"에코바이오\",\"MKT_VAL\":36434,\"WGT\":0.17,\"S_WGT\":100.67,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":30,\"TOP60\":3,\"APT_SHR_CNT\":7200493},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"000440\",\"CMP_KOR\":\"중앙에너비스\",\"MKT_VAL\":35028,\"WGT\":0.16,\"S_WGT\":100.83,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":31,\"TOP60\":3,\"APT_SHR_CNT\":1556783},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"294630\",\"CMP_KOR\":\"서남\",\"MKT_VAL\":28512,\"WGT\":0.13,\"S_WGT\":100.97,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":32,\"TOP60\":3,\"APT_SHR_CNT\":13261279},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"043200\",\"CMP_KOR\":\"파루\",\"MKT_VAL\":28207,\"WGT\":0.13,\"S_WGT\":101.10,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":33,\"TOP60\":3,\"APT_SHR_CNT\":33861495},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"065420\",\"CMP_KOR\":\"에스아이리소스\",\"MKT_VAL\":25096,\"WGT\":0.12,\"S_WGT\":101.22,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":34,\"TOP60\":3,\"APT_SHR_CNT\":49695616},{\"IDX_CD\":\"G10\",\"IDX_NM_KOR\":\"WICS 에너지\",\"ALL_MKT_VAL\":21370791,\"CMP_CD\":\"313760\",\"CMP_KOR\":\"윌링스\",\"MKT_VAL\":10963,\"WGT\":0.05,\"S_WGT\":101.27,\"CAL_WGT\":1.00,\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEQ\":35,\"TOP60\":3,\"APT_SHR_CNT\":1421917}],\"sector\":[{\"SEC_CD\":\"G25\",\"SEC_NM_KOR\":\"경기관련소비재\",\"SEC_RATE\":10.26,\"IDX_RATE\":0},{\"SEC_CD\":\"G35\",\"SEC_NM_KOR\":\"건강관리\",\"SEC_RATE\":9.34,\"IDX_RATE\":0},{\"SEC_CD\":\"G50\",\"SEC_NM_KOR\":\"커뮤니케이션서비스\",\"SEC_RATE\":7.40,\"IDX_RATE\":0},{\"SEC_CD\":\"G40\",\"SEC_NM_KOR\":\"금융\",\"SEC_RATE\":8.94,\"IDX_RATE\":0},{\"SEC_CD\":\"G10\",\"SEC_NM_KOR\":\"에너지\",\"SEC_RATE\":1.92,\"IDX_RATE\":100.00},{\"SEC_CD\":\"G20\",\"SEC_NM_KOR\":\"산업재\",\"SEC_RATE\":11.59,\"IDX_RATE\":0},{\"SEC_CD\":\"G55\",\"SEC_NM_KOR\":\"유틸리티\",\"SEC_RATE\":1.32,\"IDX_RATE\":0},{\"SEC_CD\":\"G30\",\"SEC_NM_KOR\":\"필수소비재\",\"SEC_RATE\":2.56,\"IDX_RATE\":0},{\"SEC_CD\":\"G15\",\"SEC_NM_KOR\":\"소재\",\"SEC_RATE\":8.95,\"IDX_RATE\":0},{\"SEC_CD\":\"G45\",\"SEC_NM_KOR\":\"IT\",\"SEC_RATE\":37.72,\"IDX_RATE\":0}],\"size\":[{\"SEC_CD\":\"WMI520\",\"SEC_NM_KOR\":\"WMI500 중형주\",\"SEC_RATE\":21.96,\"IDX_RATE\":82.87},{\"SEC_CD\":\"WMI530\",\"SEC_NM_KOR\":\"WMI500 소형주\",\"SEC_RATE\":78.04,\"IDX_RATE\":17.13}]}"

